dallinger/command_line/utils.py

```diff
@@ -303,13 +303,13 @@ def verify_id(ctx, param, app):
     return app
 
 
-def run_pre_launch_checks(config):
+def run_pre_launch_checks(config, mode_from_cli, cli_invocation_options: dict = {}):
```

Comment:
In your current code `locals()` which is passed to `cli_invocation_options` already contains `config` and `mode`. So, shall we simplify the signature for `run_pre_launch_checks` and `validate_config` further?

i.e. instead of this:

```py
run_pre_launch_checks(config, mode, locals())

...
def run_pre_launch_checks(config, mode_from_cli, cli_invocation_options: dict = {}):
...

def validate_config(
    self, _mode_from_cli: str = "", _cli_invocation_options: dict = {}
):
...
```

we can instead have this:

```py
run_pre_launch_checks(locals())

...
def run_pre_launch_checks(**kwargs):
    ...

def validate_config(self, **kwargs):
```